libs/parallelism/algorithms/include/hpx/parallel/algorithms/make_heap.hpp

```diff
@@ -0,0 +1,525 @@
+        Iter sequential_make_heap(
+            Iter first, Sent last, Comp&& comp, Proj&& proj)
+        {
+            using difference_type =
+                typename std::iterator_traits<Iter>::difference_type;
+
+            difference_type n = last - first;
+            if (n > 1)
+            {
+                for (difference_type start = (n - 2) / 2; start >= 0; --start)
+                {
+                    sift_down(first, comp, proj, n, first + start);
+                }
+                return first + n;
+            }
+            return first;
+        }
+
+        //////////////////////////////////////////////////////////////////////
+        template <typename Iter>
+        struct make_heap : public detail::algorithm<make_heap<Iter>, Iter>
+        {
+            make_heap()
+              : make_heap::algorithm("make_heap")
+            {
+            }
+
+            template <typename ExPolicy, typename RndIter, typename Sent,
+                typename Comp, typename Proj>
+            static RndIter sequential(
+                ExPolicy, RndIter first, Sent last, Comp&& comp, Proj&& proj)
+            {
+                return sequential_make_heap(first, last,
+                    std::forward<Comp>(comp), std::forward<Proj>(proj));
+            }
+
+            template <typename ExPolicy, typename RndIter, typename Sent,
+                typename Comp, typename Proj>
+            static
+                typename util::detail::algorithm_result<ExPolicy, RndIter>::type
+                make_heap_thread(ExPolicy&& policy, RndIter first, Sent last,
+                    Comp&& comp, Proj&& proj)
+            {
+                typename std::iterator_traits<RndIter>::difference_type n =
+                    last - first;
+                if (n <= 1)
+                {
+                    return util::detail::algorithm_result<ExPolicy,
+                        RndIter>::get(std::move(first));
+                }
+
+                using execution_policy = typename std::decay<ExPolicy>::type;
+                using parameters_type =
+                    typename execution_policy::executor_parameters_type;
+                using executor_type = typename execution_policy::executor_type;
+
+                using scoped_executor_parameters =
+                    util::detail::scoped_executor_parameters_ref<
+                        parameters_type, executor_type>;
+
+                scoped_executor_parameters scoped_params(
+                    policy.parameters(), policy.executor());
+
+                std::vector<hpx::future<void>> workitems;
+                std::list<std::exception_ptr> errors;
+
+                using tuple_type = hpx::util::tuple<RndIter, std::size_t>;
+
+                auto op = [=](tuple_type const& t) {
+                    sift_down_range(first, comp, proj, (std::size_t) n,
+                        hpx::util::get<0>(t), hpx::util::get<1>(t));
```

Comment:
```suggestion
                using tuple_type = hpx::tuple<RndIter, std::size_t>;

                auto op = [=](tuple_type const& t) {
                    sift_down_range(first, comp, proj, (std::size_t) n,
                        hpx::get<0>(t), hpx::get<1>(t));
```
Looks like the include with `hpx/modules/datastructures.hpp` or `hpx/datastructures/tuple.hpp` is missing. I also noticed yesterday that the deprecation warnings were not enabled for 1.6.0, and I've started updating names that I missed in the previous deprecation PRs (there are quite a few, so I recommend you don't enable it here).

Reply:
Correct, I'll add the missing header file. Thanks for noticing!